modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/attachment/AttachmentManagerImpl.java

```diff
@@ -153,17 +154,26 @@ public FileEntry getOrAddFileEntry(
 			serviceContext.getUserId());
 
 		try (InputStream inputStream = new ByteArrayInputStream(fileContent)) {
+			String title = DLUtil.getUniqueTitle(
+				groupId, dlFolder.getFolderId(),
+				FileUtil.stripExtension(fileName));
+
+			String sourceFileName = DLUtil.getUniqueFileName(
+				groupId, dlFolder.getFolderId(), fileName, true);
+			String mimeType = _mimeTypes.getContentType(inputStream, fileName);
```

Comment:
`_mimeTypes.getContentType` may consume the input stream. Can you double check that `getContentType` resets the mark to the beginning of the input stream? If not, you will need to do it yourself, otherwise some prefix of the input stream may be missing in the saved file.

Reply:
Alternatively, you can use the input stream only for getting the content type, and use the DLApp method that takes an array instead.